rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,169 @@
+# RFD 109: Opsgenie Integration
+## Required approvers
+
+Engineering: @r0mant, @marcoandredinis, @hugoShaka
+Product: @klizhentas, @xinding33
+Security: @reedloden, @jentfoo
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly. 
+Doing so by creating a new Teleport service 'plugin_service' with the opsgenie plugin (and any future ones) being configurable resources. This will allow the plugins to be run either by themselves or alongside other Teleport services.
+
+## Scope
+
+The plugin will be able to run standalone or alongside other services as part of the new plugin_service.
+The plugin will support auto-approval similar to PagerDuty plugin.
+
+## Success criteria
+
+### Plugin service
+The creation of a Teleport service where plugins can be created as configurable resources similar to the existing 'app_service' and 'db_service'.
+
+Plugins managed through this service should have the ability to be enabled and disabled dynamically via 'tctl'.
+
+### Opsgenie plugin
+Users are able to configure teleport to automatically create alerts in Opsgenie from access requests.
+Alerts were chosen over incidents in Opsgenie as incidents are intended to be used for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call.
+
+## Configuration UX
+
+The plugin service will be configured using the existing Teleport YAML file in a section called 'plugin_service'.
+
+```
+plugin_service:
+    enabled: true
+    plugins:
+    - "type": "opsgenie"
+    - "type": "pagerduty"
+    opsgenie:
+        api_key: "path/to/key.txt" # Path to a file containing the Opsgenie API key
+```
+This example would only match plugin resources with the the labels 'type:opsgenie', or 'type:slack'.
+
+The Opsgenie plugin (and any others created) can then be configured using resources. 
```

Comment:
Can you add a section about IaC as well? Basically, say that Terraform provider, Kubernetes operator will be updated to support managing the Plugin resource and Helm charts will be updated to support plugin_service.

And add it to the tracking ticket too please.